skutil/preprocessing/tests/test_impute.py

```diff
@@ -144,6 +144,7 @@ def test_selective_imputer():
     assert_fails(SelectiveImputer(def_fill=['a', 'b', 'c']).fit, TypeError, a)
     assert_fails(SelectiveImputer(def_fill='a').fit, TypeError, a)
     assert_fails(SelectiveImputer(def_fill=[1, 2, 'a']).fit, TypeError, a)
+    assert_fails(SelectiveImputer().fit, TypeError, a)
```

Comment:
Ok I see what you're trying to do. You assumed the `None` would propagate up, but you didn't see this line: 
```python
self.fill_ = self._def_fill if def_fill is None else def_fill
```
In a future edit, I'll fix that line to be less confusing (will make `'mean'` the default arg in subclasses). Don't worry about it for now... Here's how I would test it. Feel free to steal this snippet and try it (and then blame me if it fails):

```python
class SomeObject(object):
    def __init__(self):
        pass

assert_fails(SelectiveImputer(def_fill=SomeObject()).fit, TypeError, a)
```